samples/interop/NativeEmbedSample/iOS/EmbedSample.iOS.cs

```diff
@@ -0,0 +1,65 @@
+#if IOS
+using System;
+using System.IO;
+using System.Diagnostics;
+using Avalonia.Controls.Platform;
+using Avalonia.Platform;
+using CoreGraphics;
+using Foundation;
+using UIKit;
+using WebKit;
+
+namespace NativeEmbedSample;
+
+public partial class EmbedSample
+{
+    private IPlatformHandle CreateIOS(IPlatformHandle parent)
+    {
+        if (IsSecond)
+        {
+            var webView = new WKWebView(CGRect.Empty, new WKWebViewConfiguration());
+            webView.LoadRequest(new NSUrlRequest(new NSUrl("https://www.apple.com/")));
+
+            return new UIViewHandle(webView);
+        }
+        else
+        {
+            var button = new UIButton();
+            var clickCount = 0;
+            button.SetTitle("Hello world", UIControlState.Normal);
+            button.BackgroundColor = UIColor.Blue;
+            button.AddTarget((_, _) =>
+            {
+                clickCount++;
+                button.SetTitle($"Click count {clickCount}", UIControlState.Normal);
+            }, UIControlEvent.TouchDown);
+
+            return new UIViewHandle(button);   
+        }
+    }
+
+    private void DestroyIOS(IPlatformHandle control)
+    {
+        base.DestroyNativeControlCore(control);
+    }
+}
+
+internal class UIViewHandle : INativeControlHostDestroyableControlHandle
+{
+    private UIView _view;
+
+    public UIViewHandle(UIView view)
+    {
+        _view = view;
+    }
+
+    public IntPtr Handle => _view?.Handle ?? IntPtr.Zero;
```

Comment:
Since we are running with Xamarin bindings, we should expose UIViewHandle from the backend itself and expose a wrapped UIView as a property. Using native handles is error prone in this case, IMO.

Reply:
We still need native handles for custom IPlatformHandle interface implementations.
But yes, normally users should use typed implementation from Avalonia.iOS/Android. I already added it before, but forgot to use in the samples.